From 23f0fe0168dc493a91fa2ae8e1456d8d89daa5e2 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 21 Aug 2018 22:43:17 +0800
Subject: [PATCH 4433/5242] MLK-19260-2: spi: correct rx watermark level

commit  fdfc26a88518a711b3c0bdd57b9c5da9284ce8cc from
https://source.codeaurora.org/external/imx/linux-imx.git

Below description copied from RM:
"RX_THRESHOLD. This field defines the FIFO threshold that triggers
 a RX DMA/INT request.A RX DMA/INT request is issued when the number
of data entries in the RXFIFO is greater than RX_THRESHOLD."
Then, the RX_THRESHOLD of ECSPIx_DMAREG should be set as 'wml-1',
otherwise, the last 'wml' number of data in rxfifo will never be pulled
out.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/spi/spi-imx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-imx.c b/drivers/spi/spi-imx.c
index 61cb919..d486812 100644
--- a/drivers/spi/spi-imx.c
+++ b/drivers/spi/spi-imx.c
@@ -597,7 +597,7 @@ static int mx51_ecspi_config(struct spi_device *spi)
 		tx_wml = spi_imx->wml;
 
 	if (spi_imx->usedma)
-		writel(MX51_ECSPI_DMA_RX_WML(spi_imx->wml) |
+		writel(MX51_ECSPI_DMA_RX_WML(spi_imx->wml - 1) |
 			MX51_ECSPI_DMA_TX_WML(tx_wml) |
 			MX51_ECSPI_DMA_RXT_WML(spi_imx->wml) |
 			MX51_ECSPI_DMA_TEDEN | MX51_ECSPI_DMA_RXDEN |
-- 
1.7.9.5

